File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_146_A.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class codeforces_146_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();
        String input = scanner.nextLine();
        List<Integer> number = new ArrayList<>();
        for (char c : input.toCharArray()) {
            number.add(c - '0');
        }
        int h1 = 0;
        int h2 = 0;
        int x = 0;
        for (int i = 0; i < n; i++) {
            if (number.get(i) != 4 && number.get(i) != 7) {
                System.out.println("NO");
                x += 1;
                break;
            }
            if (i < n / 2) {
                h1 += number.get(i);
            } else {
                h2 += number.get(i);
            }
        }
        if (x == 0) {
            if (h1 == h2) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: dataset/avatar/Python/Code/codeforces_146_A.py
n = int ( input ( ) )
number = list ( map ( int , list ( input ( ) ) ) )
h1 = 0
h2 = 0
x = 0
for i in range ( n ) :
    if number [ i ] != 4 and number [ i ] != 7 :
        print ( "NO" )
        x += 1
        break
    if i < n // 2 :
        h1 += number [ i ]
    else :
        h2 += number [ i ]
if not x :
    if h1 == h2 :
        print ( "YES" )
    else :
        print ( "NO" )

